path_table/src/lib.rs

```diff
@@ -17,9 +17,25 @@ pub struct PathTable<R> {
     wildcard: Option<Box<Wildcard<R>>>,
 }
 
+#[derive(Copy, Clone, PartialEq)]
+enum WildcardCountModifier {
```

Comment:
It might make sense to rename this enum to `WildcardKind` with the variants of `Segment` and `Catchall` or something akin to this. `Segment` would be used for path segments and `Catchall` would be used for the catchall mechanism you've written here.